### ISSUE TITLE
Time zone issue with Pacific Time (UTC-8h)

### ISSUE THREAD

**Issue by fabriciomurta** [bridgedotnet/Bridge#3773]
Crossing the daylight saving thru date/time operations results in strange inconsistency. At least if the time is being added to, it is not behaving like in native .NET, and it may not be an browser-specific issue (how browsers handle daylight saving, etc).

The issue is reproducible with the dates provided below if the time zone in the running computer is set to *US Pacific time (UTC-08:00)*.

### Steps To Reproduce

https://dev.deck.net/9c74587a962cc00f96797bea39e57f23

Console application code:

```csharp
public class Program
{
    public static void Main()
    {
        var step = new TimeSpan(0, 15, 0);
        var dateUtc = new DateTime(2018, 11, 4, 7, 45, 0, DateTimeKind.Utc);
        var utcFormat = "yyyy MM dd hh:mm:ss tt";
        var localFormat = "yyyy MM dd hh:mm:ss tt zzz";

        for (var i = 0; i < 20; i++)
        {
            Console.WriteLine("utc: " + dateUtc.ToString(utcFormat) + " local: " + dateUtc.ToLocalTime().ToString(localFormat));
            dateUtc += step;
        }
    }
}
```
### Expected Result

```
utc: 2018 11 04 07:45:00 AM -08:00 local: 2018 11 04 12:45:00 AM -07:00
utc: 2018 11 04 08:00:00 AM -08:00 local: 2018 11 04 01:00:00 AM -07:00
utc: 2018 11 04 08:15:00 AM -08:00 local: 2018 11 04 01:15:00 AM -07:00
utc: 2018 11 04 08:30:00 AM -08:00 local: 2018 11 04 01:30:00 AM -07:00
utc: 2018 11 04 08:45:00 AM -08:00 local: 2018 11 04 01:45:00 AM -07:00
utc: 2018 11 04 09:00:00 AM -08:00 local: 2018 11 04 01:00:00 AM -08:00
utc: 2018 11 04 09:15:00 AM -08:00 local: 2018 11 04 01:15:00 AM -08:00
utc: 2018 11 04 09:30:00 AM -08:00 local: 2018 11 04 01:30:00 AM -08:00
utc: 2018 11 04 09:45:00 AM -08:00 local: 2018 11 04 01:45:00 AM -08:00
utc: 2018 11 04 10:00:00 AM -08:00 local: 2018 11 04 02:00:00 AM -08:00
utc: 2018 11 04 10:15:00 AM -08:00 local: 2018 11 04 02:15:00 AM -08:00
utc: 2018 11 04 10:30:00 AM -08:00 local: 2018 11 04 02:30:00 AM -08:00
utc: 2018 11 04 10:45:00 AM -08:00 local: 2018 11 04 02:45:00 AM -08:00
utc: 2018 11 04 11:00:00 AM -08:00 local: 2018 11 04 03:00:00 AM -08:00
utc: 2018 11 04 11:15:00 AM -08:00 local: 2018 11 04 03:15:00 AM -08:00
utc: 2018 11 04 11:30:00 AM -08:00 local: 2018 11 04 03:30:00 AM -08:00
utc: 2018 11 04 11:45:00 AM -08:00 local: 2018 11 04 03:45:00 AM -08:00
utc: 2018 11 04 12:00:00 PM -08:00 local: 2018 11 04 04:00:00 AM -08:00
utc: 2018 11 04 12:15:00 PM -08:00 local: 2018 11 04 04:15:00 AM -08:00
utc: 2018 11 04 12:30:00 PM -08:00 local: 2018 11 04 04:30:00 AM -08:00
```

Notice how it shows `local: 2018 11 04 01:00:00 AM` just twice, once per time zone offset.

### Actual Result

```
utc: 2018 11 04 7:45:00 AM Z local: 2018 11 04 12:45:00 AM -07:00
utc: 2018 11 04 8:00:00 AM Z local: 2018 11 04 1:00:00 AM -07:00
utc: 2018 11 04 8:15:00 AM Z local: 2018 11 04 1:15:00 AM -07:00
utc: 2018 11 04 8:30:00 AM Z local: 2018 11 04 1:30:00 AM -07:00
utc: 2018 11 04 8:45:00 AM Z local: 2018 11 04 1:45:00 AM -07:00
utc: 2018 11 04 9:00:00 AM Z local: 2018 11 04 1:00:00 AM -07:00
utc: 2018 11 04 9:15:00 AM Z local: 2018 11 04 1:15:00 AM -07:00
utc: 2018 11 04 9:30:00 AM Z local: 2018 11 04 1:30:00 AM -07:00
utc: 2018 11 04 9:45:00 AM Z local: 2018 11 04 1:45:00 AM -07:00
utc: 2018 11 04 10:00:00 AM Z local: 2018 11 04 1:00:00 AM -08:00
utc: 2018 11 04 10:15:00 AM Z local: 2018 11 04 1:15:00 AM -08:00
utc: 2018 11 04 10:30:00 AM Z local: 2018 11 04 1:30:00 AM -08:00
utc: 2018 11 04 10:45:00 AM Z local: 2018 11 04 1:45:00 AM -08:00
utc: 2018 11 04 11:00:00 AM Z local: 2018 11 04 2:00:00 AM -08:00
utc: 2018 11 04 11:15:00 AM Z local: 2018 11 04 2:15:00 AM -08:00
utc: 2018 11 04 11:30:00 AM Z local: 2018 11 04 2:30:00 AM -08:00
utc: 2018 11 04 11:45:00 AM Z local: 2018 11 04 2:45:00 AM -08:00
utc: 2018 11 04 12:00:00 PM Z local: 2018 11 04 3:00:00 AM -08:00
utc: 2018 11 04 12:15:00 PM Z local: 2018 11 04 3:15:00 AM -08:00
utc: 2018 11 04 12:30:00 PM Z local: 2018 11 04 3:30:00 AM -08:00
```

Notice occurrence of `local: 2018 11 04 1:00:00 AM` three times. Twice for -7h timezone offset.

### See Also

- Bridge forums thread: [DateTime.ToLocalTime() not functioning as expected](https://forums.bridge.net/forum/bridge-net-pro/bugs/5966)

**Comment by geoffreymcgill** [comment]
The browsers appear to handle adding over daylight saving time differently. 

Run the following in a browser console with the timezone set to Pacific Time (UTC-8h).

```js
var d = new Date(Date.UTC(2018, 10, 4, 7, 45, 0, 0, 1));

for (var i = 0; i < 20; i++) {
    d.setMinutes(d.getMinutes() + 15);
    console.log(d);
}
```

```
Sun Nov 04 2018 01:00:00 GMT-0700 (Pacific Daylight Time)
Sun Nov 04 2018 01:15:00 GMT-0700 (Pacific Daylight Time)
Sun Nov 04 2018 01:30:00 GMT-0700 (Pacific Daylight Time)
Sun Nov 04 2018 01:45:00 GMT-0700 (Pacific Daylight Time)
Sun Nov 04 2018 02:00:00 GMT-0800 (Pacific Standard Time) <-- .NET = 01:00:00 AM -08:00
Sun Nov 04 2018 02:15:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 02:30:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 02:45:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 03:00:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 03:15:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 03:30:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 03:45:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 04:00:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 04:15:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 04:30:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 04:45:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 05:00:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 05:15:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 05:30:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 05:45:00 GMT-0800 (Pacific Standard Time)
```

I'll try and figure out a workaround.

**Comment by geoffreymcgill** [comment]
I was able to reproduce properly using the following:

```js
var d = new Date(Date.UTC(2018, 10, 4, 7, 45, 0, 0));
var ms;

for (var i = 0; i < 20; i++) {
    ms = d.getTime() + (15 * 60 * 1000);
    d.setTime(ms);
    console.log(d);
}
```
Sun Nov 04 2018 01:00:00 GMT-0700 (Pacific Daylight Time)
Sun Nov 04 2018 01:15:00 GMT-0700 (Pacific Daylight Time)
Sun Nov 04 2018 01:30:00 GMT-0700 (Pacific Daylight Time)
Sun Nov 04 2018 01:45:00 GMT-0700 (Pacific Daylight Time)
Sun Nov 04 2018 01:00:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 01:15:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 01:30:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 01:45:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 02:00:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 02:15:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 02:30:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 02:45:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 03:00:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 03:15:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 03:30:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 03:45:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 04:00:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 04:15:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 04:30:00 GMT-0800 (Pacific Standard Time)
Sun Nov 04 2018 04:45:00 GMT-0800 (Pacific Standard Time)
```

I will fix in the **datetime-fixes** branch today.

**Comment by geoffreymcgill** [comment]
This scenario has been fixed. After another review and final testing, the fix will be included in the upcoming Bridge 17.8.0 release.

**Comment by fabriciomurta** [comment]
A full test over all time zones have been undegone and all looks good, so the fix for this will be available next Bridge release!
